File: Python/Euler_3.py
def find_max_prime_factor(number):
    factor = 2
    last_factor = 1
    while number > 1:
        if number % factor == 0:
            last_factor = factor
            number //= factor
            while number % factor == 0:
                number //= factor
        factor += 1
    return last_factor


print(find_max_prime_factor(600851475143))
print(find_max_prime_factor(129))

File: Java/Euler_3.java
public class Euler_3 {

    public static void main(String[] args){
        int current_factor = 2;
        int last_factor = 1;
        long number = 600851475143L;
        while(number > 1){
            if(number % current_factor == 0){
                last_factor = current_factor;
                number /= current_factor;
                while(number % current_factor == 0){
                    number /= current_factor;
                }
            }
            current_factor += 1;
        }
        System.out.println(current_factor);
    }
}
